### PR TITLE
upgrade pre-commit tools in docker

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

- python36 support pre-commit<=2.1.1 
image

- update cuda11.2 source image public key to avoid GPG problem, see [here](https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/)
image

### CODE REVIEW

**Comment by tianshuo78520a** [comment]
LGTM

File: tools/dockerfile/Dockerfile.ubuntu18 [hunk 1]
```diff
@@ -45,8 +50,7 @@ RUN wget -q https://cmake.org/files/v3.16/cmake-3.16.0-Linux-x86_64.tar.gz && ta
 ENV PATH=/home/cmake-3.16.0-Linux-x86_64/bin:$PATH
 
 
-RUN apt-get update && \
-  apt-get install -y python2.7 python2.7-dev \
+RUN apt-get install -y --fix-missing python2.7 python2.7-dev \
```
Comment (pangyoki) [inline tools/dockerfile/Dockerfile.ubuntu18]: 这里为什么要这样改？需要把apt-get update删掉吗？

Comment (betterpig) [reply tools/dockerfile/Dockerfile.ubuntu18:53]: 之前跑CINN流水线时到这步找不到包，我按照报错指示加上了 `--fix-missing`。
image
删掉apt-get update是因为前面已经 [update](https://github.com/PaddlePaddle/Paddle/blob/45fa8bc9551fb7505f492e881952f3a7cb5001c3/tools/dockerfile/Dockerfile.ubuntu18#L28) 过了。

Comment (betterpig) [reply tools/dockerfile/Dockerfile.ubuntu18]: 已去掉修改。

**Review by zhhsplendid — COMMENTED** [review]
LGTM

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by pangyoki — APPROVED** [review]
LGTM

**Pull request merged**